github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/publishers/impl/GitHubPRBuildStatusPublisher.java

```diff
@@ -90,6 +90,12 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnul
         String statusMsgValue = getStatusMsg().expandAll(run, listener);
         String buildUrl = publishedURL + run.getUrl();
 
+        if (isNull(c)) {
```

Comment:
the idea was that if you configured it to work and it can't work, than fail...

Reply:
NPE is a rough way to fail. Including a message as to why it's failing would be helpful. In the case I'm using this, I have a job configured to trigger on either a PR or a force push. In the case of a force push, this blows up because there is no PR cause.

Any thoughts on having an option to continue without failing if there is no PR cause?

Reply:
NPE should be fixed of course... I thought some plugin provides wrapper for steps.

Reply:
Ok, I'll take a look a little more closely to see if I can come up with something.

Reply:
I remember conditional build step plugin, it had ability to place step into step with conditionals... That what i mean under handler in comment

Reply:
I did actually use the flexible publish plugin, but it feels a bit hacky. I had to use a regex check `.+` for the input `${ENV,var="GITHUB_PR_HEAD_SHA"}` to check whether or not there was a PR cause.

Reply:
We can add conditional case in this plugin with optional dependency if they have extension point, that should look good in job-dsl code